### PR TITLE
Itm 589

### PR DESCRIPTION
**Pull request by dereknop**
First, run the[ ingest pr](https://github.com/NextCenturyCorporation/itm-ingest/pull/43).

Go to http://localhost:3000/text-based-results and confirm that you can view the DRE results. You should be able to navigate between MRE and DRE. Verify that the 'Participants' tab download is functional

### CODE REVIEW

**Comment by dereknop** [comment]
I forgot to mention, there were also some changes to the survey results page. Go to http://localhost:3000/survey-results and click on 'View Tabulated Data'. You should see that version 4 is checked by default. It should be reading whatever version is in the `.env` file

**Comment by brianpippin** [comment]
> I forgot to mention, there were also some changes to the survey results page. Go to http://localhost:3000/survey-results and click on 'View Tabulated Data'. You should see that version 4 is checked by default. It should be reading whatever version is in the `.env` file

While I'm thinking of it, we should make a future ticket where that value is a value in the database, and it can be changed on the admin page, in case we want to switch survey versions on the fly without redeploying.

**Comment by dereknop** [comment]
> While I'm thinking of it, we should make a future ticket where that value is a value in the database, and it can be changed on the admin page, in case we want to switch survey versions on the fly without redeploying.

I have a ticket for this (we had discussed earlier). It just got pushed aside by all things DRE

**Review by phile-caci — APPROVED** [review]
LGTM, Can switch between DRE/MRE; text/charts/participants work, can download results from participants. No issues in browser console.

File: dashboard-ui/src/components/TextBasedResults/TextBasedResultsPage.jsx [hunk 1]
```diff
@@ -299,122 +312,131 @@ function ParticipantView({ data, scenarioName }) {
...
+                                    qs.push(tmpResponses[scenario][q].question);
                                 }
                             }
                         }
-
                     }
+                } else {
+                    console.log(scenario)
+                    console.log(filteredTextBasedConfigs)
+                    console.log(filteredTextBasedConfigs[scenario])
```
Comment (kaitlyn-sharo) [inline dashboard-ui/src/components/TextBasedResults/TextBasedResultsPage.jsx:431]: Maybe remove these?

**Review by kaitlyn-sharo — COMMENTED** [review]
We've already discussed handling NaN%s better than I originally did, but just mentioning that here for others reviewing. 

In the MRE scenarios, I'm seeing an "undefined" option at the top of every set
image

Also, when I choose MRE, all the DRE scenarios still show up in the list and can be chosen.

While trying to view the participant charts in a DRE scenario, I got this:
image

File: dashboard-ui/src/components/TextBasedResults/TextBasedResultsPage.jsx [hunk 2]
```diff
@@ -185,81 +180,99 @@ function SingleGraph({ data, pageName }) {
...
-                const time_key = key + ' time (s)';
-                if (typeof (page[key]) === 'object' && !Array.isArray(page[key])) {
-                    if (!headers.includes(time_key)) {
-                        headers.push(time_key);
-                    }
+            // Check if page is defined before using Object.keys
+            if (page) {
+                Object.keys(page).forEach(key => {
+                    if (key === 'alignmentData') {
+                        console.log('hit')
```
Comment (kaitlyn-sharo) [inline dashboard-ui/src/components/TextBasedResults/TextBasedResultsPage.jsx:224]: cleanup!

**Comment by nextcen-dgemoets** [comment]
When I downloaded the data for `DryRunEval.MJ1`, there seemed to be some duplicate rows, e.g., for patients 20249202 (2 entries), 20249204 (4 entries), 20249205 (3 entries), 20249206 (2 entries), 20249210 (3 entries), and 20249212 (2 entries).

Spot-checking, I seemed to get similar behavior for other ADEPT scenarios, but not ST.

**Comment by dereknop** [comment]
> When I downloaded the data for `DryRunEval.MJ1`, there seemed to be some duplicate rows, e.g., for patients 20249202 (2 entries), 20249204 (4 entries), 20249205 (3 entries), 20249206 (2 entries), 20249210 (3 entries), and 20249212 (2 entries).
> 
> Spot-checking, I seemed to get similar behavior for other ADEPT scenarios, but not ST.

Looking into this and DRE scenarios showing up under MRE selection (can't reproduce so far). Resolved undefined options

**Comment by nextcen-dgemoets** [comment]
Not sure if this has been reported (or the charts are still TBD), but many of the chart titles appear to be off, as if it's a `split` call gone wrong, e.g., "After inject, After inject, After inject, After inject, What action do you take?" in `qol-dre-1-eval`.

**Comment by nextcen-dgemoets** [comment]
So...the stacktrace I was getting yesterday (that prompted me to re-import my DB) has resurfaced.
After replacing my DB, I did the import and dashboard update for ITM-589.  Now when I switch to `development` or `ITM-585_UpdateResultsPageforDRE`, I get the below stacktrace on the home page.  If I got back to `ITM-589`, it works again.

This is the stacktrace:
```
Cannot read properties of undefined (reading 'toLowerCase')
TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at http://localhost:3000/static/js/bundle.js:6804:113
    at Array.find (<anonymous>)
    at App.setupTextBasedConfig (http://localhost:3000/static/js/bundle.js:6804:61)
    at children (http://localhost:3000/static/js/bundle.js:6845:16)
    at Query (http://localhost:3000/static/js/bundle.js:22005:31)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:319628:22)
    at updateFunctionComponent (http://localhost:3000/static/js/bundle.js:323195:24)
    at beginWork (http://localhost:3000/static/js/bundle.js:324914:20)
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:309884:18)
    at Object.invokeGuardedCallbackDev (http://localhost:3000/static/js/bundle.js:309928:20)
```

The dev console also shows:

```
The above error occurred in the <Query> component:

    at Query (http://localhost:3000/static/js/bundle.js:22327:24)
    at Router (http://localhost:3000/static/js/bundle.js:338049:30)
    at App (http://localhost:3000/static/js/bundle.js:6725:5)
    at ApolloProvider (http://localhost:3000/static/js/bundle.js:22211:19)
    at Provider (http://localhost:3000/static/js/bundle.js:614498:3)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```

**Comment by dereknop** [comment]
> So...the stacktrace I was getting yesterday (that prompted me to re-import my DB) has resurfaced. After replacing my DB, I did the import and dashboard update for ITM-589. Now when I switch to `development` or `ITM-585_UpdateResultsPageforDRE`, I get the below stacktrace on the home page. If I got back to `ITM-589`, it works again.
> 
> This is the stacktrace:

Yea this error is from how we were matching images to patients in the survey and text based scenarios. This branch accounts for the new files but dev won't yet. I would say merge this branch into dev locally so you can get the page to work. Using the index.jsx from this branch will work if you don't want to merge the full branch in

**Comment by dereknop** [comment]
> 20249206

I looked in my mongo compass and confirmed that there are in fact duplicate documents with the same PID. It would seem that some results are being uploaded more than once. I have to look through them manually but so far the choices all seem the same so it really is just the same document being uploaded. Not an issue with this page but I need to investigate this behavior

**Review by kaitlyn-sharo — APPROVED** [review]
Everything looks fixed to me!

**Review by kaitlyn-sharo — COMMENTED** [review]
Although I will say that I am no longer seeing "After inject" in the MRE data, which should be there. Not sure how much the correct presentation of MRE data matters now, but I remember that being _very_ important back then

**Pull request merged**